src/deck.ts

```diff
@@ -0,0 +1,90 @@
+type Suit = "spades" | "hearts" | "diamonds" | "clubs"
+
+type Rank = 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14
+
+type ordering = 1 | 0 | -1
+            // GT | EQ| LT
+
+interface Card {
+    rank: Rank
+    suit: Suit
+}
+
+function cardCompare(card1: Card, card2: Card): ordering {
+    if (card1.rank > card2.rank) {
+        return 1
+    } else if (card1.rank < card2.rank) {
+        return -1
+    } else return 0
+}
+
+function cardCompareTrump(card1: Card, card2: Card, trump: Suit = "spades"): ordering {
+    if (card1.suit == trump) {
+        return 1
+    } else if (card2.suit == trump) {
+        return -1
+    } else return cardCompare(card1, card2)
+}
+
+
+class Deck {
+    deck: Card[]
+    hands: Card[][]
+    trumpCard: Card
+    trump: Suit
+
+    constructor(players: number, startingCard: Rank = 6) {
+        // * add all of the cards from the deck
+        // * give each player 6 random cards
+        // * pull random remaining card for trump
+        this.deck = this.createDeck(startingCard)
+        this.hands = this.deal(players)
+        this.trumpCard = this.finalDraw()
```

Comment:
you're assigning the value of trumpCard both on this line, and inside the finalDraw() function itself, if you want to keep trumpCard, remove one of the assignments